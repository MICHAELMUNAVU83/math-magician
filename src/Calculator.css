@import url('https://fonts.googleapis.com/css2?family=Crete+Round&family=Inter:wght@200;300&family=Josefin+Sans:wght@100&family=Kanit:ital,wght@1,300;1,400&family=Kdam+Thmor+Pro&family=Roboto+Mono:ital,wght@1,500&display=swap');
.calcApp {
  display: grid;
  justify-content: center;
  align-content: center;
  min-height: 90vh;
  grid-template-columns: repeat(4, 100px);
  grid-template-rows: minmax(100px, auto) repeat(5, 80px);
  background-color: rgba(255, 255, 255, 0.75);
}
.fullcalc{
    display: flex;
    flex-direction: row;
    justify-content: space-around;
}
p{
    font-size: 40px;
    font-family: 'Roboto Mono', monospace;
}

.calcApp > button {
  cursor: pointer;
  outline: none;
  border: 1px solid white;
}

.operand {
  background-color: orange;
}

.span-two {
  grid-column: span 2;
}

.output {
  grid-column: 1 / -1;
  display: flex;
  align-items: flex-end;
  justify-content: space-around;
  background-color: grey;
  flex-direction: column;
  color: white;
  padding: 10px;
  word-wrap: break-word;
  word-break: break-all;
}
