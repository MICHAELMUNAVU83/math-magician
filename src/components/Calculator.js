import React , {useState } from "react";
import calculate from "../logic/calculate";

const Calculator = () => {
  const [calc, setCalc] = useState({
    total: 0,
    next: "",
    operation: "",
  });

  const clickCalc = (e) => {
    const values = e.target.textContent;
    setCalc((obj) => calculate(obj, values));
  };

  const results = `
    ${
      calc.total || calc.operation || calc.next
        ? `${calc.total || ""} ${calc.operation || ""} ${calc.next || ""}`
        : 0
    }`;

  return (
    <div className="calcApp">
      <div className="output">
        {/* <div className="previous"></div> */}
        <div className="current">{results}</div>
      </div>

      <button type ="button" onClick ={clickCalc} >AC</button>
      <button type ="button" onClick ={clickCalc}>+/-</button>
      <button type ="button" onClick ={clickCalc} >%</button>
      <button type ="button" onClick ={clickCalc} className="operand" >
        รท
      </button>
      <button type ="button" onClick ={clickCalc} >7</button>
      <button type ="button" onClick ={clickCalc} >8</button>
      <button type ="button" onClick ={clickCalc} >9</button>
      <button type ="button" onClick ={clickCalc}  className="operand">
        x
      </button>
      <button type ="button" onClick ={clickCalc} >4</button>
      <button type ="button" onClick ={clickCalc} >5</button>
      <button type ="button" onClick ={clickCalc} >6</button>
      <button type ="button" onClick ={clickCalc} className="operand" >
        -
      </button>
      <button type ="button" onClick ={clickCalc} >1</button>
      <button type ="button" onClick ={clickCalc} >2</button>
      <button type ="button" onClick ={clickCalc} >3</button>
      <button type ="button"  onClick ={clickCalc} className="operand">
        +
      </button>
      <button  type ="button" onClick ={clickCalc} className="span-two">
        0
      </button>
      <button type ="button" onClick ={clickCalc} >.</button>
      <button type ="button" onClick ={clickCalc} className="operand" >
        =
      </button>
    </div>
  );
};

export default Calculator;
