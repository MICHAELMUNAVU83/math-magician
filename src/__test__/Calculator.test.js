import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('Calculator renders correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
