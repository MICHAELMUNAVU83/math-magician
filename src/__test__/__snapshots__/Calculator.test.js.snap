// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
<div
  className="fullcalc"
>
  <div>
    <p>
      Lets do some math
    </p>
  </div>
  <div
    className="calcApp"
  >
    <div
      className="output"
    >
      <div
        className="current"
      >
        
    0
      </div>
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="operand"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="operand"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="operand"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="operand"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="span-two"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="operand"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
